crypto/stack/stack.c

```diff
@@ -167,6 +171,14 @@ static int sk_reserve(OPENSSL_STACK *st, int n, int exact)
     num_alloc = st->num + n;
     if (num_alloc < min_nodes)
         num_alloc = min_nodes;
+    
+    if (st->data == NULL) {  /* Lazy allocation */
+        st->data = OPENSSL_zalloc(sizeof(void *) * num_alloc);
+        if (st->data == NULL)
+            return 0;
+        st->num_alloc = num_alloc;
+        return 1;
+    }
```

Comment:
??? num_alloc is not ready computed at this point. It should be choice between OPENSSL_zalloc and OPENSSL_realloc further below...

Reply:
Well, it does work, but placement is somewhat counter-intuitive. And if you move it compute_growth goes to infinite loop. So just moving won't work...

Reply:
Basically choice is between moving (and working about endless loop) and clarifying commentary.

Reply:
I choose to add a few comments.